qemu/tests/remote_server_disconnected.py

```diff
@@ -0,0 +1,70 @@
+import os
+import netaddr
+import logging
+
+from avocado.utils import process
+
+
+def run(test, params, env):
+    """
+    Convert remote image.
+
+    1) Start VM
+    2) Disconnect local host from the remote server one by one,
+       make sure the vm can be accessed.
+    """
+
+    def _check_hosts(hosts):
+        if len(hosts) < 2:
+            test.cancel("2 remote servers at least are required.")
+        for h in hosts:
+            if os.path.exists(h) or netaddr.valid_ipv6(h):
+                test.cancel("Neither ipv6 nor unix domain"
+                            " socket is supported by now.")
+
+    hosts = []
+    if params.get("enable_gluster") == "yes":
+        hosts.append(params["gluster_server"])
+        hosts.extend(params.get("gluster_peers", "").split())
+
+    _check_hosts(hosts)
+    hosts.pop()  # The last server should be accessible
+
+    disconn_cmd = params["disconn_cmd"]
+    recover_cmd = params["recover_cmd"]
+    conn_check_cmd = params["conn_check_cmd"]
+    disk_op_cmd = params["disk_op_cmd"]
+    disk_op_tm = int(params["disk_op_timeout"])
+
+    try:
+        session = None
+        disconn_hosts = []
+
+        vm = env.get_vm(params["main_vm"])
+        vm.verify_alive()
+        tm = int(params.get("login_timeout", 300))
+        session = vm.wait_for_login(timeout=tm)
+
```

Comment:
better to put try here, the smaller, the better.

Reply:
U are right, but what will happen if something wrong with verify_alive or wait_for_login, when an exception is raised by those function calls, tc breaks, but there is no way to clean up the testing env if we put try after wait_for_login.

Reply:
if an error is raised by one of `get_vm`, `verify_alive`, `wait_for_login`, there is no need to recover the connection and the variable `session` is `None` in all of those circumstances.

Reply:
Please consider such scenario, qemu-kvm is running, when boot os inside VM, something wrong occurs, boot fails, then the VM os hang by showing a user shell (# ) to let user handle the error, in this situation, VM is alive, but wait_for_login must fail, so, finally, we have to call vm.destroy() to kill the qemu-kvm process, what's your opinion please?

Reply:
this is a valid point, simply put `kill_vm = yes` could solve this.

Reply:
added kill_vm = yes in cfg, and move try right before the for loop, thanks for your comments, Longxiang.